File: 351002/Lashchenka/publisher/src/main/java/by/bsuir/distcomp/controller/AuthController.java
package by.bsuir.distcomp.controller;

import by.bsuir.distcomp.dto.request.LoginRequestTo;
import by.bsuir.distcomp.dto.response.LoginResponseTo;
import by.bsuir.distcomp.service.AuthService;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v2.0")
public class AuthController {

    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/login")
    public ResponseEntity<LoginResponseTo> login(@Valid @RequestBody LoginRequestTo request) {
        return ResponseEntity.ok(authService.login(request));
    }
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/controller/AuthController.kt
package com.example.distcomp.controller

import com.example.distcomp.dto.request.LoginRequestTo
import com.example.distcomp.dto.response.AuthTokenResponseTo
import com.example.distcomp.dto.response.CurrentUserResponseTo
import com.example.distcomp.service.AuthService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v2.0")
class AuthController(
    private val authService: AuthService
) {
    @PostMapping("/login")
    fun login(@Valid @RequestBody request: LoginRequestTo): AuthTokenResponseTo =
        authService.login(request)

    @GetMapping("/me")
    fun currentUser(): CurrentUserResponseTo = authService.currentUser()
}
